lib/portlayer/storage/nfs/volume.go

```diff
@@ -272,3 +274,37 @@ func (v *VolumeStore) getMetadata(op trace.Operation, ID string, target Target)
 	op.Infof("Successfully read volume metadata at (%s)", metadataPath)
 	return info, nil
 }
+
+func (v *VolumeStore) StatPath(op trace.Operation, deviceId string, targetPath string) (*compute.FileStat, error) {
+	target, err := v.Service.Mount(op)
+	if err != nil {
+		return nil, err
+	}
+	defer v.Service.Unmount(op)
+
+	files, err := target.ReadDir(v.volDirPath(deviceId))
+	var result string
+	for _, file := range files {
+		result = result + " " + file.Name()
+	}
+
+	fileInfo, _, err := target.Lookup(path.Join(v.volDirPath(deviceId), targetPath))
+	if err != nil {
+		return nil, err
+	}
+
+	// nfs server returns "" for file name when doing a look up, need to fill in the value from targetPath instead
+	// trimsuffix shouldn't be needed since the statpath personality should trim the suffix
+	newPath := targetPath
+	for strings.HasSuffix(newPath, "/") {
+		newPath = strings.TrimSuffix(newPath, "/")
+	}
+	// length of pathTokens is guaranteed to be >= 1
+	pathTokens := strings.Split(strings.TrimSuffix(newPath, "/"), "/")
+	fileName := pathTokens[len(pathTokens) - 1]
+
+	// FIXME: cannot read symlink.
```

Comment:
Talked to George about this. I need to make a ticket for implementing `ReadLink` in the go-nfs-client for this. Most likely during the hardening time if possible.